src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProvider.cs

```diff
@@ -64,6 +64,8 @@ public ModelTypeProvider(InputModelType inputModel, string defaultNamespace, Sou
             _defaultDerivedType = defaultDerivedType ?? (inputModel.IsDefaultDiscriminator ? this : null);
         }
 
+        private static string GetDefaultNamespace(string defaultNamespace)
```

Comment:
Can we consolidate this into one method?

Reply:
Yes, I want to. But I cannot find an common place to provide this.

Reply:
Why this cannot be in the emitter side?

Reply:
No. emitter will not do such calculation/compute, it will emitter what is defined in cadl or configuration